src/WindowsPackageManager/main.cpp

```diff
@@ -58,6 +61,23 @@ extern "C"
     }
     CATCH_RETURN();
 
+    void WINDOWS_PACKAGE_MANAGER_API_CALLING_CONVENTION WindowsPackageManagerServerWilResultCallback(const wil::FailureInfo& failure) noexcept try
+    {
+        AppInstaller::Logging::Telemetry().LogFailure(failure);
+    }
+    CATCH_LOG();
+
+    WINDOWS_PACKAGE_MANAGER_API WindowsPackageManagerServerCreateInstance(const CLSID* clsid, const IID* iid, void** out) try
+    {
+        RETURN_HR_IF_NULL(E_POINTER, clsid);
+        RETURN_HR_IF_NULL(E_POINTER, iid);

```

Comment:
The 2 input args should return E_INVALIDARG if null

Reply:
Or even better, use REFCLSID and REFIID like line 101 so that you don't even need to check null